src/main/java/frc/robot/commands/auto/AimForNote.java

```diff
@@ -2,35 +2,64 @@
 
 import com.chaos131.swerve.BaseSwerveDrive;
 
+import edu.wpi.first.math.MathUtil;
 import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants;
+import frc.robot.subsystems.Feeder;
+import frc.robot.subsystems.Intake;
 import frc.robot.subsystems.Vision;
 import frc.robot.subsystems.Vision.CameraDirection;
+import frc.robot.subsystems.launcher.Launcher;
+import frc.robot.subsystems.vision.CameraInterface;
+import frc.robot.subsystems.vision.CameraInterface.CameraMode;
 
 public class AimForNote extends Command {
 	private BaseSwerveDrive m_swerveDrive;
 	private Vision m_vision;
+	private Intake m_intake;
+	private CameraInterface m_camera;
+	private final double m_speed = -0.5; // needs to be in range [-1.0, 1.0] negative values are backwards
+	private Launcher m_launcher;
+	private Feeder m_feeder;
+	private double m_startime;
+	private final double m_minDurationSeconds = 1.0;
 
-	public AimForNote(BaseSwerveDrive swerveDrive, Vision vision){
+	public AimForNote(BaseSwerveDrive swerveDrive, Vision vision, Intake intake, Launcher launcher, Feeder feeder){
 		m_swerveDrive = swerveDrive;
+		m_intake = intake;
 		m_vision = vision;
-		addRequirements(swerveDrive, vision);
+		m_camera = m_vision.getCamera(CameraDirection.Back);
+		m_launcher = launcher;
+		m_feeder = feeder;
+		addRequirements(swerveDrive, intake, vision, launcher, feeder);
 	}
 
 	public void initialize() {
-		// 
+		m_vision.getCamera(CameraDirection.Back).setMode(CameraMode.PIECE_TRACKING);
+		m_startime = Timer.getFPGATimestamp();
+		m_intake.setIntakePower(1);
 	}
 
-	
-
 	/** The main body of a command. Called repeatedly while the command is scheduled. */
 	public void execute() {
-		Double tx = m_vision.getCamera(CameraDirection.Back).getTargetAzimuth(true);
-		if(!m_vision.getCamera(CameraDirection.Back).hasTarget()) return;
+		m_feeder.grabAndHoldPiece(0.35);
+		m_launcher.setTiltAngle(Constants.LauncherConstants.VisionIntakeAngle);
+
+		if (!m_camera.isCorrectPipeline()) {
+			// lets protect ourselves from targetting an april tag and moving to it
+			return;
+		}
+		Double tx = m_camera.getTargetAzimuth(true);
+		Double ty = m_camera.getTargetElevation(true);
+		if(!m_camera.hasTarget()) return;
 
 		Rotation2d noteAngle = m_swerveDrive.getOdometryRotation().rotateBy( Rotation2d.fromDegrees(-tx) );
-		// TODO: Drive into the note until it is intaken (intook?)
-		m_swerveDrive.moveFieldRelativeAngle(0, 0, noteAngle, 0.6);
+		var speedModifier = MathUtil.clamp(ty/50, 0.1, 1.0);
```

Comment:
We tried this quickly, but was too slow.
THe command also ended before it got to the note (maybe it lost the target)

Reply:
That sucks. Did it only run for 1s, or did it get close to the note and end the command?

Reply:
Sorry! Just saw your question

It may have run for one second, but started doing a sidesweeping motion and may have lost the note

Reply:
The model we used wasn't giving us good results in the room and we set the confidence was pretty low to get it to recognize any notes. Better lighting or a different model might help. Maybe something else caught its attention.

It also needs a debouncer-like tracker or memory structure so it doesn't flit around like a goldfish.